frontend/src/pages/hardwareProfiles/utils.ts

```diff
@@ -1,4 +1,229 @@
+  allInvalid: boolean,
+  hasInvalid: boolean,
+  allIncomplete: boolean,
+): string => {
+  if (allInvalid) {
+    return HardwareProfileBannerWarningTitles.ALL_INVALID;
+  }
+  if (!hasEnabled) {
+    return HardwareProfileBannerWarningTitles.ALL_DISABLED;
+  }
+  if (hasInvalid) {
+    return HardwareProfileBannerWarningTitles.SOME_INVALID;
+  }
+  if (allIncomplete) {
+    return HardwareProfileBannerWarningTitles.ALL_INCOMPLETE;
+  }
+  return HardwareProfileBannerWarningTitles.SOME_INCOMPLETE;
+};
+
+export const generateWarningMessage = (
+  hasEnabled: boolean,
+  allInvalid: boolean,
+  hasInvalid: boolean,
+  allIncomplete: boolean,
+): string => {
+  if (allInvalid) {
+    return 'You must have at least one valid hardware profile enabled for users to create workbenches or deploy models. Take the appropriate actions below to re-validate your profiles.';
+  }
+  if (!hasEnabled) {
+    return 'You must have at least one hardware profile enabled for users to create workbenches or deploy models. Enable one or more profiles in the table below.';
+  }
+  if (hasInvalid) {
+    return 'One or more of your defined hardware profiles are invalid. Take the appropriate actions below to revalidate your profiles.';
+  }
+  if (allIncomplete) {
+    return 'All of your defined hardware profiles are missing either CPU or Memory. This is not recommended.';
+  }
+  return 'One or more of your defined hardware profiles are missing either CPU or Memory. This is not recommended.';
+};
+
+export const generateWarningForHardwareProfiles = (
+  hardwareProfiles: HardwareProfileKind[],
+): WarningNotification | undefined => {
+  const hasInvalid = hardwareProfiles.some((profile) => {
+    const warning = hardwareProfileWarning(profile);
+    return (
+      typeof warning !== 'undefined' &&
+      !warning.message.includes(HARDWARE_PROFILES_MISSING_CPU_MEMORY_MESSAGE)
+    );
+  });
+  const hasEnabled = hardwareProfiles.some((profile) => profile.spec.enabled);
+  const allInvalid = hardwareProfiles.every((profile) => {
+    const warning = hardwareProfileWarning(profile);
+    return (
+      typeof warning !== 'undefined' &&
+      !warning.message.includes(HARDWARE_PROFILES_MISSING_CPU_MEMORY_MESSAGE)
+    );
+  });
+  const allIncomplete = hardwareProfiles.every((profile) => {
+    const warning = hardwareProfileWarning(profile);
+    return (
+      typeof warning !== 'undefined' &&
+      warning.message.includes(HARDWARE_PROFILES_MISSING_CPU_MEMORY_MESSAGE)
+    );
+  });
+  const hasInvalidIncludingIncompleteProfiles = hardwareProfiles.some(
+    (profile) => typeof hardwareProfileWarning(profile) !== 'undefined',
+  );
+  if (hardwareProfiles.length === 0 || (!hasInvalidIncludingIncompleteProfiles && hasEnabled)) {
+    return undefined;
+  }
+  return {
+    title: generateWarningTitle(hasEnabled, allInvalid, hasInvalid, allIncomplete),
+    message: generateWarningMessage(hasEnabled, allInvalid, hasInvalid, allIncomplete),
+  };
+};
```

Comment:
Now this function after updating the other func (hardwareProfileWarning) you can instead look for the specific warnings like
```tsx
const warnings = hardwareProfileWarning(profile);
const invalid === warnings.some(warning => warning.type === HARDWARE_PROFILES_MISSING_CPU_MEMORY)
```